### PR TITLE
first version includes tests and browser example

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "eslint": "^3.9.1",
+    "eslint-config-firstandthird": "^3.0.2",
+    "eslint-plugin-import": "^2.1.0",
+    "hapi": "^13.4.1",
+    "lab": "^10.8.2"
+  },
+  "eslintConfig": {
+    "extends": "firstandthird"
+  },
+  "dependencies": {}
```
Comment (jgallen23) [inline package.json]: missing lodash.  Can we use aug instead?

File: index.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  varietiesToInclude: ['view'],
+  fetchDirectives: {
+    'default-src': ['https:'],
+    'report-uri': 'http://localhost/csp_reports'
+  }
+};
+
+exports.register = (server, pluginOptions, next) => {
+  const options = _.defaults(pluginOptions, pluginDefaults);
+  const headerKey = 'Content-Security-Policy-Report-Only';
```
Comment (jgallen23) [inline index.js]: We should make this configurable

**Review by jgallen23 — CHANGES_REQUESTED** [review]

File: test/test.mixed.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+});
+
+lab.afterEach((done) => {
+  server.stop(() => {
+    done();
+  });
+});
+
+
+lab.test('should add default header to incoming requests of the indicated variety ', (allDone) => {
```
Comment (jgallen23) [inline test/test.mixed.js:24]: Let's add a test to make sure the report route is set up

File: index.js (hunk 2 above)
Comment (orthagonal) [reply index.js]: Heh I originally had this in pluginDefaults, then somehow thought my way out of it.  Will put it back.

File: index.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
+const _ = require('lodash');
+
+const pluginDefaults = {
+  varietiesToInclude: ['view'],
+  fetchDirectives: {
+    'default-src': ['https:'],
```
Comment (jgallen23) [inline index.js:9]: Can we switch these to defaultSrc and reportUri

Comment (jgallen23) [reply index.js:9]: Oh. Nevermind

File: index.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  });
+  // will set up an endpoint at report-uri if you want:
+  if (options['report-uri']) {
+    const routeOptions = {
+      uri: options['report-uri'],
+      method: 'POST'
+    };
+    // will need to try this out in browser:
+    routeOptions.handler = options.routeHandler ? options.routeHandler : (request, reply) => {
+      server.log(['content-security-policy-report'], request.payload);
```
Comment (jgallen23) [inline index.js]: lets make these tasks configurable

**Pull request merged**